pkg/template/template.go

```diff
@@ -55,6 +55,12 @@ func NewDefaultEngine() *Engine {
 			Variables: map[string]string{
 				"version": "",
 			},
+			Validate: func(vars map[string]string) error {
```

Comment:
This becomes a little tricky when the templates are externalized(read from file) and any template could be used. Wondering if there is some kinda validation we could do in the Templates instead?

Reply:
Was pondering about that too. There's `isset` but I couldn't find if I can error from there.

Reply:
Could make a custom function if none similar enough exist

Reply:
Done in 2243bd0

Reply:
I think the If/Else option could be useful in some cases, but for the basic 'fail if missing' use case, couldn't we just have a single statement on top pr variable instead of the whole if/else construct, eg.

    {{ failIfVariableMissing "name" }}

?

Reply:
Was thinking about that but it will result in the weird function signature because for the templates the function has to return `(value, error)`. But to be honest the current `fail` is not much different :)

Reply:
Done in 73e4cc8